Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs

```diff
@@ -602,145 +614,67 @@ private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
 
             if (!SpawnMap(uid, nukeops))
             {
-                Logger.InfoS("nukies", "Failed to load map for nukeops");
+                _sawmill.Info("Failed to load map for nukeops");
                 continue;
             }
 
-            // Basically copied verbatim from traitor code
-            var playersPerOperative = nukeops.PlayersPerOperative;
-            var maxOperatives = nukeops.MaxOps;
+            //Handle there being nobody readied up
+            if (ev.PlayerPool.Count == 0)
+                continue;
 
-            // Dear lord what is happening HERE.
-            var everyone = new List<ICommonSession>(ev.PlayerPool);
-            var prefList = new List<ICommonSession>();
-            var medPrefList = new List<ICommonSession>();
-            var cmdrPrefList = new List<ICommonSession>();
-            var operatives = new List<ICommonSession>();
+            //INCONSISTENCY! Why does RulePlayerJobsAssignedEvent use an array and this uses a list?!
+            var playerPoolArray = ev.PlayerPool.ToArray();
 
-            // The LINQ expression ReSharper keeps suggesting is completely unintelligible so I'm disabling it
-            // ReSharper disable once ForeachCanBeConvertedToQueryUsingAnotherGetEnumerator
-            foreach (var player in everyone)
-            {
-                if (!ev.Profiles.ContainsKey(player.UserId))
-                {
-                    continue;
-                }
+            var commanderEligible = _antagSelection.GetEligibleSessions(playerPoolArray, nukeops.CommanderRoleProto, false);
+            var agentEligible = _antagSelection.GetEligibleSessions(playerPoolArray, nukeops.MedicRoleProto, false);
+            var operativeEligible = _antagSelection.GetEligibleSessions(playerPoolArray, nukeops.OperativeRoleProto, false);
 
-                var profile = ev.Profiles[player.UserId];
-                if (profile.AntagPreferences.Contains(nukeops.OperativeRoleProto.Id))
-                {
-                    prefList.Add(player);
-                }
-                if (profile.AntagPreferences.Contains(nukeops.MedicRoleProto.Id))
-	            {
-	                medPrefList.Add(player);
-	            }
-                if (profile.AntagPreferences.Contains(nukeops.CommanderRoleProto.Id))
-                {
-                    cmdrPrefList.Add(player);
-                }
-            }
+            var nukiesToSelect = 6;//_antagSelection.CalculateAntagNumber(_playerManager.PlayerCount, nukeops.PlayersPerOperative, nukeops.MaxOps);
 
-            var numNukies = MathHelper.Clamp(_playerManager.PlayerCount / playersPerOperative, 1, maxOperatives);
+            //Select Nukies
+            //Select Commander, priority : commanderEligible, agentEligible, operativeEligible, all players
+            var selectedCommander = _antagSelection.ChooseAntags(new List<ICommonSession>[] { commanderEligible, agentEligible, operativeEligible, ev.PlayerPool }, 1).FirstOrDefault();
```

Comment:
ChooseAntags could be refactored a bit to take the number first then variadic args so you dont need to write the new List<ICommonSession>[] bit every time

Reply:
Completely forgot params existed